File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/bus/repository/BusApplicantRepository.java
package b1nd.dodam.domain.rds.bus.repository;

import b1nd.dodam.domain.rds.bus.entity.Bus;
import b1nd.dodam.domain.rds.bus.entity.BusApplicant;
import b1nd.dodam.domain.rds.member.entity.Student;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface BusApplicantRepository extends JpaRepository<BusApplicant, Long> {
    void deleteByBus_Id(Long busId);
    List<BusApplicant> findByBus(Bus bus);

    BusApplicant findByBusAndSeat(Bus bus, int seats);

    @EntityGraph(attributePaths = {"bus"})
    Optional<BusApplicant> findByStudent(Student student);
}

File: services/service-bus/src/main/kotlin/com/b1nd/dodamdodam/bus/domain/bus/repository/BusApplicantRepository.kt
package com.b1nd.dodamdodam.bus.domain.bus.repository

import com.b1nd.dodamdodam.bus.domain.bus.entity.BusApplicantEntity
import com.b1nd.dodamdodam.bus.domain.bus.entity.BusEntity
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface BusApplicantRepository : JpaRepository<BusApplicantEntity, Long> {
    fun deleteAllByBusId(busId: Long)
    fun findAllByBus(bus: BusEntity): List<BusApplicantEntity>
    fun findByBusAndSeat(bus: BusEntity, seat: Int): BusApplicantEntity?

    @EntityGraph(attributePaths = ["bus"])
    fun findByUserId(userId: String): BusApplicantEntity?
}
